LayerClient/LayerClient.py

```diff
@@ -0,0 +1,509 @@
+        self.bearer_token = bearer_token
+
+    def _get_layer_headers(self):
+        """
+        Convenience method for retrieving the default set of authenticated
+        headers.
+
+        Return: The headers required to authorize ourselves with the Layer
+        platform API.
+        """
+        return {
+            'Accept': 'application/vnd.layer+json; version=1.0',
+            'Authorization': 'Bearer ' + self.bearer_token,
+            'Content-Type': 'application/json'
+        }
+
+    def _get_layer_uri(self, *suffixes):
+        """
+        Used for building Layer URIs for different API endpoints.
+
+        Parameter`suffixes`: An array of strings, which will be joined as the
+            end portion of the URI body.
+
+        Return: A complete URI for an endpoint with optional arguments
+        """
+        suffix_string = '/'.join(suffixes) if suffixes else ''
+        return 'https://api.layer.com/apps/{app_id}/{suffix}'.format(
+            app_id=self.app_uuid,
+            suffix=suffix_string,
+        )
+
+    def _raw_request(self, method, url, data=None):
+        """
+        Actually make a call to the Layer API.
+        If the response does not come back as valid, raises a
+        LayerPlatformException with the error data from Layer.
+
+        Parameters:
+        - `method`: The HTTP method to use
+        - `url`: The target URL
+        - `data`: Optional post body. Must be json encodable.
+
+        Return: Raw JSON doc of the Layer API response
+
+        Exception: `LayerPlatformException` if the API returns non-OK response
+        """
+        result = requests.request(
+            method,
+            url,
+            headers=self._get_layer_headers(),
+            data=(json.dumps(data) if data else None)
+        )
+
+        if not result.ok:
+            try:
+                error = result.json()
+                raise LayerPlatformException(
+                        error.get('message'),
+                        http_code=result.status_code,
+                        code=error.get('code'),
+                        error_id=error.get('id'),
+                )
+            except ValueError:
+                # Catches the JSON decode error for failures that do not have
+                # associated data
+                raise LayerPlatformException(
+                    result.text,
+                    http_code=result.status_code,
+                )
+
+        return result.json()
+
+    def get_conversation(self, conversation_uuid):
+        """
+        Fetch an existing conversation by UUID
+
+        Parameter `conversation_uuid`: The UUID of the conversation to fetch
+
+        Return: A `Conversation` instance
+        """
+        return Conversation.from_dict(
+            self._raw_request(
+                METHOD_GET,
+                self._get_layer_uri(
+                    LAYER_URI_CONVERSATIONS,
+                    conversation_uuid,
+                ),
```

Comment:
These were intentional, as they make diffs prettier down the line. If this
is against internal style guidelines I can change it though.
On 15 Dec 2015 16:37, "abdinoor-jana" <notifications@github.com> wrote:

> In LayerClient/LayerClient.py
> <https://github.com/Jana-Mobile/layer-python/pull/1#discussion_r47702398>:
>
> > +
> > +    def get_conversation(self, conversation_uuid):
> > +        """
> > +        Fetch an existing conversation by UUID
> > +
> > +        Parameter `conversation_uuid`: The UUID of the conversation to fetch
> > +
> > +        Return: A `Conversation` instance
> > +        """
> > +        return Conversation.from_dict(
> > +            self._raw_request(
> > +                METHOD_GET,
> > +                self._get_layer_uri(
> > +                    LAYER_URI_CONVERSATIONS,
> > +                    conversation_uuid,
> > +                ),
>
> unnecessary comma
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Jana-Mobile/layer-python/pull/1/files#r47702398>.
>